slow_odgi/slow_odgi/inject.py

```diff
@@ -31,7 +31,10 @@ def where_chop(graph, pathname, index):
             return None
         length = len(graph.segments[handle.name].seq)
         if walk + length > index:
-            return handle.name, index - walk
+            if handle.orientation:
+                return handle.name, index - walk
+            else:
+                return handle.name, length - (index - walk)
```

Comment:
Very slick! Almost makes me wish for a little method on `Handle` that does this "direction flipping" itself. Something like:

```py
def pos(self, n):
    """Get the concrete index in the underlying segment sequence corresponding
    to the `n`th nucleotide from the beginning (in the appropriate direction).
    """
    if self.orientation:
        return n
    else:
        return <somehow get the length> - n
```

But of course we don't have access to the length there, so maybe this should actually be a method on `Segment` or something.

Reply:
Ah this would be real nice, but I think we're a little trapped by our interning goals. Handles don't know about segment lengths, and segments don't know about orientations. And... these are both good things! 

You really need a global view of the graph to be able to do the above. I could lift this little bit of math into `preprocess.py`, if that's any nicer? I could go either way.

Reply:
No particular need, I say! Just something to keep in mind for the future, IMO, in case it comes up again in a solvable way.